### PR TITLE
Improvements to group name parsing

### PR DESCRIPTION
**Pull request by nirbheek**
```
Fix thinko in create_group() log message
```

```
Simplify try..except..else clauses
The `else:` case is not required since we already `return` in the
`except:`case.
```

```
Strip leading and trailing punctuations from group pings
This has been happening for a while, and most recently at:
```
https://www.reddit.com/comments/ieys9l//g2kgaqj

Fixes https://github.com/neoliberal/user_pinger/issues/11

```
Validate group names when creating and pinging
This is to ensure that we don't accidentally start using punctuation
in the group names, which will interact badly with the 'strip
punctuation' logic.

We can always expand this list later if we want to use other characters.
```

### CODE REVIEW

File: user_pinger.py [hunk 1]
```diff
@@ -80,6 +86,12 @@ def save(self) -> None:
             return
         return
 
+    def _validate_group_name(self, group_name: str) -> Tuple[str, str]:
```
[the remark this answers is not in the document]
Comment (nirbheek) [reply user_pinger.py]: Fixed, thanks!

File: user_pinger.py [hunk 2]
```diff
@@ -208,16 +220,26 @@ def handle_comment(self, comment: praw.models.Comment) -> None:
...
+            self.logger.debug("End of comment with no group specified")
+            return
+        self.logger.debug("Comment contained group token %s", token)
+
+        # Strip punctuation like:
+        # !ping WEEBS, what do you folks think of this?
+        trigger: str = token.strip(self.GROUP_STRIP_PUNCT)
+
+        # Validate group name to save having to look up invalid names
+        ret: bool, msg: Optional[str] = self._validate_group_name(trigger)
```
Comment (jenbanim) [inline user_pinger.py:236]: I get a syntax error on this line using Python 3.6.9

Based on [PEP 526](https://www.python.org/dev/peps/pep-0526/) and [this stackoverflow thread](https://stackoverflow.com/questions/52082939/type-hints-when-unpacking-a-tuple) I don't think unpacking a tuple like this is allowed in any version of python.

```suggestion
        ret: bool
        msg: Optional[str]
        ret, msg = self._validate_group_name(trigger)
```

File: user_pinger.py [hunk 3]
```diff
@@ -582,6 +604,13 @@ def create_group(body: str, author: praw.models.Redditor) -> None:
             Note:
             Moderator will be a member of the group
             """
+            group_name: str = body.upper()
+            ret: bool, msg: Optional[str] = self._validate_group_name(group_name)
```
Comment (jenbanim) [inline user_pinger.py:608]: Same issue here as above

```suggestion
            ret: bool
            msg: Optional[str]
            ret, msg = self._validate_group_name(group_name)
```

File: user_pinger.py [hunk 4]
```diff
@@ -208,16 +220,26 @@ def handle_comment(self, comment: praw.models.Comment) -> None:
...
+
+        # Strip punctuation like:
+        # !ping WEEBS, what do you folks think of this?
+        trigger: str = token.strip(self.GROUP_STRIP_PUNCT)
+
+        # Validate group name to save having to look up invalid names
+        ret: bool, msg: Optional[str] = self._validate_group_name(trigger)
+        if not ret:
+            self.logger.debug(msg)
+            return
```
Comment (jenbanim) [inline user_pinger.py:239]: Would be helpful to let people know why their ping didn't go through

```suggestion
            self._send_error_pm(f"Invalid group name", [f"The group name {trigger} contains invalid characters"], comment.author)
            return
```

Comment (jenbanim) [reply user_pinger.py:242]: Whoops, gave that the wrong description

**Pull request merged**